### PR TITLE
Poll from server

### PR DESCRIPTION
**Pull request by YonatanKra**
I've added a service that gets data from the server.
I've also changed the component to fetch data from the server and poll it every few milliseconds.
The server address and the query interval are set in the `apps/vaccitron/environments` folder. For production it should be set in the `environments.prod` file and for tests/development it is currently set to take the file from the assets folder.

### CODE REVIEW

File: apps/vaccitron/src/app/report-list-page/report-list-page.component.spec.ts [hunk 1]
```diff
@@ -1,39 +1,18 @@
...
 import { NotificationsFilter, VaccinesReport } from '@vacgaps/interfaces';
 import { By } from '@angular/platform-browser';
 import { FilterFormModule } from '@vacgaps/filter-form';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { VaccinesReportsService } from '@vacgaps/vaccines-reporter';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { of } from 'rxjs';
+import { environment } from '../../environments/environment';
+
+const MOCK_REPORTS = require('../../assets/demo-data.json');
```
Comment (idigra) [inline apps/vaccitron/src/app/report-list-page/report-list-page.component.spec.ts:14]: It's not mock anymore.

File: apps/vaccitron/src/app/report-list-page/report-list-page.component.ts [hunk 2]
```diff
@@ -46,9 +25,21 @@ export class ReportListPageComponent implements OnInit {
     this.#currentFilter = newFilter;
   }
 
-  constructor() {}
+  constructor(private vaccinesReportsService: VaccinesReportsService) {}
 
-  ngOnInit(): void {}
+  ngOnInit(): void {
+    this.getUpdate();
+    interval(environment.reportsQueryInterval)
```
Comment (idigra) [inline apps/vaccitron/src/app/report-list-page/report-list-page.component.ts]: Good for now to trigger the interval internally. But if we do it that way, we need to push an update to the UI components in order for the UI to update, where is it?
(Or that it's some Angular's dependency property pattern that I don't know?)

File: apps/vaccitron/src/assets/demo-data.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+[
```
Comment (idigra) [inline apps/vaccitron/src/assets/demo-data.json:1]: When is this demo data used?

File: apps/vaccitron/src/environments/environment.prod.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
 export const environment = {
   production: true,
+  vaccinesDataUrl: 'fakeUrl',
+  reportsQueryInterval: 5 * 60 * 1000 // 5 minutes
```
Comment (idigra) [inline apps/vaccitron/src/environments/environment.prod.ts]: reportsQueryMinutesInterval

**Review by idigra — CHANGES_REQUESTED** [review]

File: apps/vaccitron/src/app/report-list-page/report-list-page.component.spec.ts (hunk 1 above)
Comment (YonatanKra) [reply apps/vaccitron/src/app/report-list-page/report-list-page.component.spec.ts:14]: It is still a mock. This is for dev/test only.

File: apps/vaccitron/src/app/report-list-page/report-list-page.component.ts (hunk 2 above)
Comment (YonatanKra) [reply apps/vaccitron/src/app/report-list-page/report-list-page.component.ts]: What I should actually do is create a service that handles that an exposes a Subjuect that listens to server updates or interval. Let's take this as tech debt and open a task for it? Or would you like it fixed now?

File: apps/vaccitron/src/assets/demo-data.json (hunk 3 above)
Comment (YonatanKra) [reply apps/vaccitron/src/assets/demo-data.json:1]: In dev and tests. Try to run the app locally

File: apps/vaccitron/src/environments/environment.prod.ts (hunk 4 above)
Comment (YonatanKra) [reply apps/vaccitron/src/environments/environment.prod.ts]: It is in milliseconds actually. Would you like me to state it in the propname?

Comment (idigra) [reply apps/vaccitron/src/environments/environment.prod.ts]: Yes, applies to all the stack.

File: apps/vaccitron/src/app/report-list-page/report-list-page.component.ts (hunk 2 above)
Comment (idigra) [reply apps/vaccitron/src/app/report-list-page/report-list-page.component.ts]: Talked offline, as I suspected it's Binding - another type of dependency property that I don't know :)

**Review by idigra — APPROVED** [review]

**Pull request merged**